Source/EasyNetQ/RabbitAdvancedBus.cs

```diff
@@ -105,91 +105,54 @@ AdvancedBusEventHandlers advancedBusEventHandlers
-            consumer.StartConsuming();
-            return consumer;
-        }
-
-        /// <inheritdoc />
-        public IDisposable Consume<T>(
-            IQueue queue, IMessageHandler<T> onMessage, Action<IConsumerConfiguration> configure
-        )
-        {
-            Preconditions.CheckNotNull(queue, "queue");
-            Preconditions.CheckNotNull(onMessage, "onMessage");
-            Preconditions.CheckNotNull(configure, "configure");
-
-            return Consume(queue, x => x.Add(onMessage), configure);
-        }
-
-        /// <inheritdoc />
-        public IDisposable Consume(
-            IQueue queue, Action<IHandlerRegistration> addHandlers, Action<IConsumerConfiguration> configure
-        )
-        {
-            Preconditions.CheckNotNull(queue, "queue");
-            Preconditions.CheckNotNull(addHandlers, "addHandlers");
-            Preconditions.CheckNotNull(configure, "configure");
-
-            var handlerCollection = handlerCollectionFactory.CreateHandlerCollection(queue);
-            addHandlers(handlerCollection);
-
-            return Consume(queue, (body, properties, messageReceivedInfo, cancellationToken) =>
-            {
-                var deserializedMessage = messageSerializationStrategy.DeserializeMessage(properties, body);
-                var handler = handlerCollection.GetHandler(deserializedMessage.MessageType);
-                return handler(deserializedMessage, messageReceivedInfo, cancellationToken);
-            }, configure);
-        }
-
-        /// <inheritdoc />
-        public virtual IDisposable Consume(
-            IQueue queue, MessageHandler onMessage, Action<IConsumerConfiguration> configure
-        )
-        {
-            Preconditions.CheckNotNull(queue, "queue");
-            Preconditions.CheckNotNull(onMessage, "onMessage");
-            Preconditions.CheckNotNull(configure, "configure");
-
-            if (disposed)
-                throw new EasyNetQException("This bus has been disposed");
-
-            var consumerConfiguration = new ConsumerConfiguration(configuration.PrefetchCount);
-            configure(consumerConfiguration);
-            var consumer = consumerFactory.CreateConsumer(queue, (body, properties, receivedInfo, cancellationToken) =>
-            {
-                var rawMessage = produceConsumeInterceptor.OnConsume(new ConsumedMessage(receivedInfo, properties, body));
-                return onMessage(rawMessage.Body, rawMessage.Properties, receivedInfo, cancellationToken);
-            }, consumerConfiguration);
+            var consumeConfiguration = new ConsumeConfiguration(configuration.PrefetchCount);
+            configure(consumeConfiguration);
+
+            var consumerConfiguration = new ConsumerConfiguration(
+                consumeConfiguration.PrefetchCount,
+                consumeConfiguration.PerQueueConsumeConfigurations.ToDictionary(
+                    x => x.Item1,
```

Comment:
For readability, it's better to make a separate class or a `ValueTuple` wuth named elements.

Reply:
Let me think about this later on(it's internal api), I don't want to add ValueTuple dependency, but I'm a bit unsure how to name this stuff.